src/interactivity/win32/Clipboard.cpp

```diff
@@ -52,35 +52,35 @@ contents and writing them to the console's input buffer
 --*/
 void Clipboard::Paste()
 {
-    HANDLE ClipboardDataHandle;
-
-    // Clear any selection or scrolling that may be active.
-    Selection::Instance().ClearSelection();
-    Scrolling::s_ClearScroll();
+    // WIP auditing if user is enrolled
+    static auto DestinationName = _LoadString(ID_CONSOLE_WIP_DESTINATIONNAME);
+    Microsoft::Console::Internal::EdpPolicy::AuditClipboard(DestinationName);

```

Comment:
I don't trust that this is safe - it introduces a TOCTOU vulnerability because the clipboard isn't held while we check EDP applicability, right?

Reply:
Oh right, I forgot to ask you about this bit of code. I wanted to know where to put the `AuditClipboard` call since I didn't know how it worked. After all, when we're holding the clipboard lock, how would any other component possibly acquire it?

Reply:
Okay, so, EDP uses some _other_ clipboard runoff (specifically: WNF notifications) to determine the things it determines. It doesn't operate on the lock _or even the clipboard_